File: 1190-reverse-substrings-between-each-pair-of-parentheses/1190-reverse-substrings-between-each-pair-of-parentheses.py
class Solution:
    def reverseParentheses(self, s: str) -> str:
        stack = []
        for c in s:
            if c == ")":
                reverse = []
                while stack[-1] != "(":
                    reverse.append(stack.pop())
                stack.pop()
                stack += reverse
            else:
                stack.append(c)
        return "".join(stack)

File: 1190-reverse-substrings-between-each-pair-of-parentheses/1190-reverse-substrings-between-each-pair-of-parentheses.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseParentheses = function(s) {
    const stack = []
    for (const c of s) {
        if (c === ")") {
            let reversed = ""
            while (stack.at(-1) !== "(") {
                reversed += stack.pop()
            }
            stack.pop()
            stack.push(...reversed)
        } else {
            stack.push(c)
        }
    }
    return stack.join("")
};